### DISCUSSION TITLE
Markdown to PDF without using LaTeX as intermediate format

### DISCUSSION THREAD

**Discussion by johnmapeson** [jgm/pandoc#9753] [Q&A] [answered]
I need to convert Markdown to PDF without using LaTeX as intermediate format.

For this, I tried

```
pandoc input.md --pdf-engine=wkhtmltopdf -o output.pdf
```

but it turns out I need to install Rosetta to make it work on Mac, and moreover, `wkhtmltopdf` has been depreacted. I'm not sure I want to install Rosetta. Which PDF engine I can use instead of `wkhtmltopdf`? Or maybe there is a completely different way, a workaround?

**Comment by tarleb** [comment, marked as answer]
In short, [possible values for pdf-engine](https://pandoc.org/MANUAL#option--pdf-engine) are:

- pdfroff – *extremely* fast, but output is not as polished
- prince – only non-opensource solution, but free for personal use; high quality output
- weasyprint – very good quality converter written in Python
- pagedjs-cli – printing via Chrome, required a local JavaScript dev environment last time I checked.
- typst – the newest, and in some ways, shiniest tool in the list; quite fast, good quality, uses new tech.

Plus the ones you mentioned, and ConTeXt, which is not LaTeX but still a TeX-based solutions.

If you care about PDF/A support, then that only leaves prince, context, and weasyprint, but weasyprint doesn't have full support, I believe.

**Reply by johnmapeson** [reply]
Thanks, Albert!
